### PR TITLE
Box plot changes - Part 1

### PR DESCRIPTION
**Pull request by bhanvimenghani**
Adds box plot chart on top of the recent changes in mvp_demo. Fixes tooltip layout and display.

### CODE REVIEW

**Comment by kusumachalasani** [comment]
@bhanvimenghani Can you please share the image ?

File: src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationComponents/PerfDetails.tsx [hunk 1]
```diff
@@ -22,7 +22,7 @@ interface Alert {
   type: AlertType;
 }
 
-const PerfDetails = (props: { recommendedData; currentData; chartData; day; endtime; displayChart; tab }) => {
+                                                                      const PerfDetails = (props: { recommendedData; currentData; chartData; day; endtime; displayChart; tab; boxPlotData }) => {
```
Comment (kusumachalasani) [inline src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationComponents/PerfDetails.tsx:25]: Remove the extra spaces.

File: src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationTables.tsx [hunk 2]
```diff
@@ -96,11 +109,67 @@ const RecommendationTables = (props: { endTimeArray; setEndTimeArray; SREdata; s
...
+              const yData = containerData.y.map((data) => [data.min, data.q1, data.median, data.q3, data.max]);
+  
+              return {
+                name: 'cpu',
+                x: timestamps,
+                y: yData
+              };
+            });
+  
+            // for box plots data points
```
Comment (kusumachalasani) [inline src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationTables.tsx:141]: follow the same pattern to comment or describe the section of code.
No extra line between comments and code.

File: src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationTables.tsx [hunk 3]
```diff
@@ -96,11 +109,67 @@ const RecommendationTables = (props: { endTimeArray; setEndTimeArray; SREdata; s
...
+  
+              return {
+                name: 'cpu',
+                x: timestamps,
+                y: cpuDetails
+              };
+            } else return 0;
+          });
+  
+          // console.log(chartData);
```
Comment (kusumachalasani) [inline src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationTables.tsx:127]: Remove unwanted commented out code.

File: src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationTables.tsx [hunk 4]
```diff
@@ -96,11 +109,67 @@ const RecommendationTables = (props: { endTimeArray; setEndTimeArray; SREdata; s
         setCurrentData(current_arr);
         setRecommendedData(recommended_arr);
         setChartDetailsData(chartDetailsObject);
-      }
-    };
+          // setBoxPlotData(boxPlot_arr);
```
Comment (kusumachalasani) [inline src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationTables.tsx:112]: Do we need this? if not, remove the unwanted commented out code.

File: src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationTables.tsx [hunk 5]
```diff
@@ -48,23 +51,33 @@ const RecommendationTables = (props: { endTimeArray; setEndTimeArray; SREdata; s
       if (endtime && day) {
         const response = await fetch(list_recommendations_url);
         const result = await response.json();
+        const boxPlot_arr: any = [];
         const recommended_arr: any = [];
         const current_arr: any = [];
         const chartDetailsObject = [];
 
+                // current data
+
```
Comment (kusumachalasani) [inline src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationTables.tsx:60]: follow the same pattern to comment or describe the section of code.
No extra line between comments and code. Commented line should inline with code.

File: src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationTables.tsx [hunk 6]
```diff
@@ -23,14 +23,17 @@ const RecommendationTables = (props: { endTimeArray; setEndTimeArray; SREdata; s
...
   const [currentData, setCurrentData] = useState([]);
   const [recommendedData, setRecommendedData] = useState([]);
   const [chartDetailsData, setChartDetailsData] = useState([]);
   const [day, setDay] = useState('short_term');
   const [displayChart, setDisplayChart] = useState(true);
 
+
+  const [boxPlotTranslatedData, setBoxPlotTranslatedData] = useState([]);
+
+
```
Comment (kusumachalasani) [inline src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationTables.tsx:36]: Remove the extra lines.

Comment (bhanvimenghani) [reply src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationTables.tsx:36]: done

File: src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationTables.tsx (hunk 5 above)
Comment (bhanvimenghani) [reply src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationTables.tsx:60]: done

File: src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationTables.tsx (hunk 4 above)
Comment (bhanvimenghani) [reply src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationTables.tsx:112]: these r debugging statements, in the other pr these r mostly removed

File: src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationTables.tsx (hunk 3 above)
Comment (bhanvimenghani) [reply src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationTables.tsx:127]: this one is also a debugging statement, I can remove it now also

File: src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationTables.tsx (hunk 2 above)
Comment (bhanvimenghani) [reply src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationTables.tsx:141]: done

File: src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationComponents/PerfDetails.tsx (hunk 1 above)
Comment (bhanvimenghani) [reply src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationComponents/PerfDetails.tsx:25]: done

**Comment by kusumachalasani** [comment]
With  quay.io/bmenghan/pr185, when there are no cpu recommendations and has only memory recommendations,
I don't see anything in the plots.

image

**Comment by bhanvimenghani** [comment]
It is not showing the box plots only, and the lin charts dont have enough data to be plotted is what i guess

**Comment by bhanvimenghani** [comment]
can u share the recommendations json

**Review by dinogun — APPROVED** [review]
Merging to get a PoC image for testing

**Pull request merged**